src/simulators/statevector/qubitvector.hpp

```diff
@@ -670,6 +673,17 @@ std::complex<data_t> QubitVector<data_t>::operator[](uint_t element) const {
   return data_[element];
 }
 
+template <typename data_t>
+void QubitVector<data_t>::set_state(uint_t pos, std::complex<data_t>& val) {
+  data_[pos] = val;
+}
+
+template <typename data_t>
+std::complex<data_t> QubitVector<data_t>::get_state(uint_t pos) const {
+  return data_[pos];
+}
```

Comment:
Isn't it enough with the `operator[](int i)` functions?

Reply:
Issue here was with thrust backend. It's `[]` operator returns raw pointer to thrust compex type and requires `get_state` to cast to std::complex. So rather than mess around with template overloads I just added the same function to the non-thrust qubitvector.